Python/specialize.c

```diff
@@ -864,7 +864,10 @@ specialize_dict_access(
         return 0;
     }
     _PyAttrCache *cache = (_PyAttrCache *)(instr + 1);
-    if (type->tp_flags & Py_TPFLAGS_INLINE_VALUES && _PyObject_InlineValues(owner)->valid) {
+    if (type->tp_flags & Py_TPFLAGS_INLINE_VALUES &&
+        _PyObject_InlineValues(owner)->valid &&
+        (base_op == LOAD_ATTR || _PyObject_GetManagedDict(owner) == NULL))
```

Comment:
The intent here is not clear (the code is correct, just unclear).
What we're saying is "don't do this specialization for stores if the dict is non-NULL". 
```suggestion
        !(base_op == STORE_ATTR && _PyObject_GetManagedDict(owner) != NULL))
```

Plus, should we ever somehow use this for another instruction, like `DELETE_ATTR`, the code will remain correct